helpers.php

```diff
@@ -0,0 +1,367 @@
+ *
+ * @param int $uniqueId
+ *
+ * @return bool|Cart
+ */
+
+function getCartFromUniqueId($uniqueId)
+{
+    $results = Db::getInstance()->getRow('SELECT id_cart 
+    FROM `' . _DB_PREFIX_ . 'altapay_transaction` 
+    WHERE unique_id=\'' . $uniqueId . '\'');
+    $cart    = new Cart((int)$results['id_cart']);
+
+    return $cart;
+}
+
+/**
+ * Use the unique ID to fetch the created from it (if any)
+ *
+ * @param int $uniqueId
+ *
+ * @return Order
+ */
+
+function getOrderFromUniqueId($uniqueId)
+{
+    $results = Db::getInstance()->getRow('SELECT id_order 
+    FROM `' . _DB_PREFIX_ . 'altapay_order` 
+    WHERE unique_id = \'' . $uniqueId . '\'');
+    $order   = new Order((int)$results['id_order']);
+
+    return $order;
+}
+
+/**
+ * Method for marking and updating status of order as captured in case of a captured action in database
+ *
+ * @param int   $paymentId
+ * @param array $orderlines
+ *
+ * @return bool
+ */
+function markAsCaptured($paymentId, $orderlines = array())
+{
+    $sql = 'UPDATE ' . _DB_PREFIX_ . 'altapay_order SET requireCapture = 0 WHERE payment_id = ' . $paymentId
+           . ' LIMIT 1';
+    Db::getInstance()->Execute($sql);
+
+    foreach ($orderlines as $productId => $quantity) {
+        if ($quantity == 0) {
+            continue;
+        }
+
+        $result = Db::getInstance()->getRow('SELECT captured 
+            FROM ' . _DB_PREFIX_ . 'altapay_orderlines WHERE altapay_payment_id = "'
+                                            . $paymentId . '" AND product_id = ' . $productId);
+
+        if (isset($result['captured'])) {
+            $quantity         += $result['captured'];
+            $sqlUpdateCapture = 'UPDATE ' . _DB_PREFIX_ .
+                                'altapay_orderlines SET captured = ' . $quantity .
+                                ' WHERE altapay_payment_id = ' . $paymentId;
+            Db::getInstance()->Execute($sqlUpdateCapture);
+        } else {
+            $sqlOrderLine = 'INSERT INTO ' . _DB_PREFIX_ .
+                            'altapay_orderlines (altapay_payment_id, product_id, captured) 
+                VALUES("' . $paymentId . '", "' . $productId . '", ' . $quantity . ')';
+            Db::getInstance()->Execute($sqlOrderLine);
+        }
+    }
+
+    return true;
+}
+
+/**
+ * Method for marking and updating status of order as refund in case of a refund action in database
+ *
+ * @param int   $paymentId
+ * @param array $orderlines
+ *
+ * @return bool
+ */
+function markAsRefund($paymentId, $orderlines = array())
+{
+    $sqlRequireCapture = 'SELECT requireCapture 
+    FROM ' . _DB_PREFIX_ . 'altapay_order WHERE payment_id = ' . $paymentId;
+    $result            = Db::getInstance()->getRow($sqlRequireCapture);
+    // Only payments which have been captured/partial captured will be considered
+    if (!isset($result['requireCapture']) || $result['requireCapture'] != 0) {
+        return false;
+    }
+    if (count($orderlines) > 0) {
```

Comment:
You do not need to check if an array is empty before giving it to a foreach loop, the for each will iterate 0 times